compiler/src/iree/compiler/Codegen/SPIRV/SPIRVVectorizeLoadStore.cpp

```diff
@@ -739,6 +744,37 @@ class PassThroughConversion : public OpConversionPattern<OpT> {
   }
 };
 
+// Helper function to optionally predicate a scalar load/store if there is a
+// mask present.
+static Value predicateMaybeMaskedScalarTransfer(
+    OpBuilder &b, Location loc, Value maybeMaskBit, TypeRange resultTypes,
+    function_ref<Value(OpBuilder &, Location)> thenConditionBuilder,
+    function_ref<void(OpBuilder &, Location)> elseConditionBuilder,
+    bool hasElse) {
+  OpBuilder::InsertionGuard guard(b);
+  if (maybeMaskBit) {
+    OpBuilder::InsertionGuard guard1(b);
+    scf::IfOp ifOp;
+    if (hasElse) {
+      ifOp = b.create<scf::IfOp>(loc, resultTypes, maybeMaskBit, /*else=*/true);
+    } else {
+      ifOp = b.create<scf::IfOp>(loc, maybeMaskBit, /*else=*/false);
+    }
```

Comment:
Isn't there a create function that handles the condition builders? I remember using one in vector gather lowering: https://github.com/llvm/llvm-project/blob/767ce07c2d4c312188623441979d620b9708ee99/mlir/lib/Dialect/Vector/Transforms/LowerVectorGather.cpp#L156C3-L160

Reply:
I couldn't find a version of the condition builders that let the else condition be optional. I can take another look.

Reply:
after closer inspection, the else builder defaults to nullptr; I'll use that instead.